cidc_api/services/files.py

```diff
@@ -5,17 +5,53 @@
 
 from eve import Eve
 from eve_sqlalchemy import parse
-from flask import Request, _request_ctx_stack
-from werkzeug.exceptions import BadRequest
+from flask import Request, request, _request_ctx_stack, Blueprint, jsonify
+from werkzeug.exceptions import BadRequest, NotFound
 from werkzeug.datastructures import ImmutableMultiDict
 
 import gcloud_client
-from models import Users, Permissions, CIDCRole
+from auth import requires_auth
+from models import Users, Permissions, CIDCRole, DownloadableFiles
+
+files_api = Blueprint("files", __name__, url_prefix="/downloadable_files")
+
+
+@files_api.route("/download_url", methods=["GET"])
+@requires_auth("download_url")
+def get_download_url():
+    """
+    Get a signed GCS download URL for a given file.
+    """
+    # Extract file ID from route
+    file_id = request.args.get("id")
+    if not file_id:
+        raise BadRequest("Missing expected URL parameter `id`")
+
+    # Check that file exists
+    file_record = DownloadableFiles.find_by_id(file_id)
+    if not file_record:
+        raise NotFound(f"No file with id {file_id}.")
+
+    user = _request_ctx_stack.top.current_user
+
+    # Ensure user has permission to access this file
+    perms = Permissions.find_for_user(user)
+    if (
+        not any(
+            perm.assay_type == file_record.assay_type
+            and perm.trial_id == file_record.trial_id
+            for perm in perms
+        )
+        and user.role != CIDCRole.ADMIN.value
```

Comment:
for the sake of readability I suggest rewrite this into nested `if`s form or something.